cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
> There is already a sensitive default of '3 days ago' when you create a stratum1 from a garbage collectible repository

It wasn't there when I checked 11 days ago, but I do see that you added it now because that's the way that the GC gets enabled on the replica.  In any case, if somebody converts an existing repo to be garbage collectible and then somebody converts the corresponding existing replica, it has to be added by hand.  That part's not so bad, it can be in the instructions, but I do still object to the double purpose for the one variable.   It's confusing, and there's no reason for most people to set a timespan.

> We see the garbage collection as an experimental feature addressing mainly the nightly integration build use-case for now. The implementation and configuration of this features will almost certainly be subject to change in future versions. I suggest that we should first gain some experience with it and later see which changes are really necessary.

The problem is that once you've got some usage in production you've always got to be backward compatible.  That constrains what you can change, especially things in configuration files.